src/stores/gameHistory.js

```diff
@@ -28,8 +28,7 @@ export const useGameHistoryStore = defineStore('gameHistory', () => {
     if (
       isNaN(index) || 
       !isFinite(index) || 
-      index < 0 || 
-      index >= gameStates.value.length
```

Comment:
Annoyingly I removed this constraint to deal with the reauthenticating case, because you'll have the query param with `gameStateIndex` but not have any gameStates yet since your request was denied. We want to be able to grab the gameStateIndex from the url (ideally via this computed prop) before we've actually got the gameStates in place. That means we could have weirdness with out-of-bounds indices, but let's cross that bridge later